library/bdBank/XenForo/DataWriter/DiscussionMessage/Post.php

```diff
@@ -18,14 +18,12 @@ public function bdBank_doBonus()
         $point = 0;
         if (bdBank_AntiCheating::checkPostQuality($this)) {
             $bonusType = $this->_bdBank_isDiscussionFirstMessage() ? 'thread' : 'post';
-            $point = $bank->getActionBonus($bonusType, array('forum' => $forum));
+            $point = $bank->getActionBonus($bonusType, $this->get('post_date'), array('forum' => $forum));
         }
 
         if (!$this->isInsert()) {
-            $reverseResult = $bank->reverseSystemTransactionByComment($comment, $point);
-            if (count($reverseResult['skipped']) > 0) {
-                return;
-            }
+            $bank->makeTransactionAdjustments($comment, $point);
```

Comment:
Let's use the old reverse+rebonus logic here. Because if a post is deleted, the old credit should be forfeited.

Reply:
done via 5feab2c95482555382fcfe35849cea64274bfb9a

Reply:
ah but then bonus adjustment for post won't grant credit.  
Need to use separate logic to rebuild bonus for posts instead of reusing `bdBank_doBonus` then?

Reply:
Or check if post is actually deleted in `bdBank_doBonus`

Reply:
fixed in 84ac3d354bf31dfc0954c9655592a5ddac5a803a

Reply:
> ah but then bonus adjustment for post won't grant credit.

That's the correct behavior I think. If user deleted the post -> all previous bonus/credit transactions are reverted. When it's undeleted -> system send out new bonus, no credit/adjustment logic is required. 

I think the changes in 84ac3d3 should be reverted.

Reply:
I'm referring to when we adjust the bonus points for post, anh.  
Example:
1 .adjust bonus for  post from 2 to 1
2. rebuild bonuses
3. if we reverse old transaction and create new one with 1 point, user won't get 1 point of credit

Reply:
if the post is soft-deleted, `checkPostQuality` would return false.  
if the post is hard-deleted, rebuild bonuses won't process the post

Reply:
ad17c95

Reply:
Hmm, maybe for rebuilding, we don't need to check `XenForo_Model_Post:: isDeleted`. Just adjust it to `$point`, always?

Reply:
ah right, reverting ad17c95 now